ipc/cli/src/commands/subnet/create.rs

```diff
@@ -112,11 +118,15 @@ pub struct CreateSubnetArgs {
         long,
         help = "The permission mode for the subnet: collateral, federated and static"
     )]
+    // TODO figure out a way to use a newtype + ValueEnum, or reference PermissionMode::VARIANTS to
+    //  enumerate all variants
     pub permission_mode: String,
```

Comment:
It would be nice to use a custom parser function like [here](https://github.com/consensus-shipyard/ipc/blob/f53dfbeaadc52aec6284214083c6936f33f95d2a/fendermint/app/options/src/genesis.rs#L58-L62) instead of allowing a `String` and failing later.

Reply:
Maybe a default value as well?

Reply:
I'd advise against setting a default value because these are foundational, immutable configuration properties for the subnet, and it'd be wise for the user to set them explicitly.

Reply:
> It would be nice to use a custom parser function like [here](https://github.com/consensus-shipyard/ipc/blob/f53dfbeaadc52aec6284214083c6936f33f95d2a/fendermint/app/options/src/genesis.rs#L58-L62) instead of allowing a `String` and failing later.

clap's `ValueEnum` would be the right choice, but these types sit in the `api` crate which should not know about CLI tooling. Hence my comment about a newtype above.

Reply:
[`e817bb3` (#691)](https://github.com/consensus-shipyard/ipc/pull/691/commits/e817bb3dfc8166dd925814b5dc14de4e2bf7ce9a)

We now use `value_parser` for these fields, but we'll need to look at newtype + ValueEnum another time.